src/main/puzzle/results-label.gd

```diff
@@ -0,0 +1,64 @@
+extends RichTextLabel
+"""
+A results label which reveals its contents line by line.
+"""
+
+# unshown character which causes the output to pause for 0.2 beats
+const LULL_CHARACTER := '/'
+
+# The delay in seconds before displaying the next character
+var _pause := 0.0
+
+# timing at which messages appear. defaults to 136 bpm
+var beat_duration := 60 / 136.0
+
+# 1.0 = normal, 2.0 = fastest, 6.0 = fastestestest, 1000000.0 = fastestest
+var _text_speed := 1.0
+
+"""
+These two fields are used to track unshown text which is gradually revealed. RichTextLabel defines a
+'visible_characters' field which supports this capability, but it does not mesh well with the scroll bar or the
+'scroll_following' property. So we reimplement the concept of invisible characters our own way.
+"""
+var _unshown_text := ""
+var _unshown_index := 0
+
+func _process(delta: float) -> void:
+	# clamped to prevent underflow (leaving the game open) or large values (a malicious string with tons of pauses)
+	_pause = clamp(_pause - delta * _text_speed, -5, 5)
+	
+	var showed_character := false
+	while _unshown_index < _unshown_text.length() and _pause <= 0.0:
+		# continue showing characters until we hit a pause
+		var unshown_char := _unshown_text[_unshown_index]
+		if unshown_char == '/':
+			_pause += 0.2 * beat_duration
+		else:
+			text += unshown_char
+			
+			if unshown_char != '\n':
+				showed_character = true
+		_unshown_index += 1
+	
+	if showed_character:
+		# play a sound; repeatedly increase the pitch for a 'counting up' sound
+		$BebebeSound.volume_db = rand_range(-7.0, -12.0)
+		$BebebeSound.pitch_scale = lerp($BebebeSound.pitch_scale, 3.0, 0.004)
+		$BebebeSound.play()
+
+
+func show_text(text_with_lulls: String) -> void:
```

Comment:
Document this method; it's mostly straightforward except `text_with_lulls` warrants a comment

Reply:
Documented show_text, hide_text.